realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -351,7 +358,11 @@ public boolean remove(Object object) {
         if (managedMode && !realm.isInTransaction()) {
             throw new IllegalStateException(REMOVE_OUTSIDE_TRANSACTION_ERROR);
         }
-        return super.remove(object);
+        boolean success = super.remove(object);
+        if (success) {
+            modCount++;
```

Comment:
Is this necessary? `super.remove(object)` will call `iterator.remove()` which should do `modCount` ?